tests/serac_profiling.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include <exception>
+
+#include <gtest/gtest.h>
+
+#include "serac/infrastructure/cli.hpp"
+#include "serac/infrastructure/initialize.hpp"
+#include "serac/infrastructure/profiling.hpp"
+#include "serac/numerics/mesh_utils.hpp"
+#include <cstring>
+
+namespace serac {
+
+TEST(serac_profiling, mesh_refinement)
+{
+  // profile mesh refinement
+  MPI_Barrier(MPI_COMM_WORLD);
+  serac::profiling::initializeCaliper();
+
+  SERAC_MARK_START("LOAD_MESH");
+
+  std::string mesh_file = std::string(SERAC_REPO_DIR) + "/data/meshes/bortel_echem.e";
+
+  SERAC_MARK_END("LOAD_MESH");
+
+  auto pmesh = mesh::refineAndDistribute(buildMeshFromFile(mesh_file), 0, 0);
+
+  SERAC_MARK_LOOP_START(refinement_loop, "refinement_loop");
+  for (int i = 0; i < 2; i++) {
+    SERAC_MARK_LOOP_ITER(refinement_loop, i);
+    pmesh->UniformRefinement();
+  }
+  SERAC_MARK_LOOP_END(refinement_loop);
+
+  SERAC_SET_METADATA("mesh_file", mesh_file.c_str());
+  SERAC_SET_METADATA("number_mesh_elements", pmesh->GetNE());
+
+  // this number represents "llnl" as an unsigned integer
+  unsigned int magic_uint = 1819176044;
+  SERAC_SET_METADATA("magic_uint", magic_uint);
+
+  // decode unsigned int back into char[4]
+  std::array<char, sizeof(magic_uint) + 1> uint_string;
+  std::fill(std::begin(uint_string), std::end(uint_string), 0);
+  std::memcpy(uint_string.data(), &magic_uint, 4);
+  std::cout << uint_string.data() << std::endl;
+
+  // encode double with "llnl" bytes
+  double magic_double;
+  std::memcpy(&magic_double, "llnl", 4);
+  SERAC_SET_METADATA("magic_double", magic_double);
+
+  // decode the double and print
+  std::array<char, sizeof(magic_double) + 1> double_string;
+  std::fill(std::begin(double_string), std::end(double_string), 0);
+  std::memcpy(double_string.data(), &magic_double, 4);
+  std::cout << double_string.data() << std::endl;
+
+  EXPECT_EQ(2112, pmesh->GetNE());
```

Comment:
It is a little confusing to me to have this be the only thing actually "tested" in this file - I think it's fine to have this be a "caliper smoketest" but I'm not clear on why this particular check is here.

Reply:
Yes, maybe it's better to just make this a smoke test.